File: frontend/src/types/activity.ts
export interface ActivityLog {
  id: number;
  user_id?: number | null;
  user_name: string;
  user_avatar_color?: string | null;
  module: string;
  action_type: string;
  title: string;
  description?: string | null;
  entity_type?: string | null;
  entity_id?: number | null;
  created_at: string;
}

export interface ActivityStats {
  total_activities: number;
  activities_today: number;
  module_counts: Record<string, number>;
  most_active_member?: string | null;
}

export interface ActivityListResponse {
  items: ActivityLog[];
  total: number;
  limit: number;
  offset: number;
}

export interface PublicMember {
  id: number;
  username: string;
  display_name: string;
  avatar_color?: string;
  role: string;
}

File: backend/app/schemas/activity.py
from datetime import datetime
from typing import Optional, List, Dict
from pydantic import BaseModel

class ActivityLogBase(BaseModel):
    module: str
    action_type: str
    title: str
    description: Optional[str] = None
    entity_type: Optional[str] = None
    entity_id: Optional[int] = None

class ActivityLogCreate(ActivityLogBase):
    user_id: Optional[int] = None
    user_name: Optional[str] = None
    user_avatar_color: Optional[str] = None

class ActivityLogResponse(ActivityLogBase):
    id: int
    user_id: Optional[int] = None
    user_name: str
    user_avatar_color: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

class ActivityStatsResponse(BaseModel):
    total_activities: int
    activities_today: int
    module_counts: Dict[str, int]
    most_active_member: Optional[str] = None

class ActivityListResponse(BaseModel):
    items: List[ActivityLogResponse]
    total: int
    limit: int
    offset: int
